### PR TITLE
Improve ux around ACL token to help users avoid overwriting node/service identities

### PR DESCRIPTION
**Pull request by roncodingenthusiast**
### Description

Follow up to https://github.com/hashicorp/consul/pull/16288 
TLDR: `node-identity` and `service-identity` flags when specified without the `merge-*` flags overwrites service identities and node identities of a token. This is ambiguous to the user because we have a `no-merge` flag which seems to imply that on update we are merging. So we have been doing work to improve the ux by renaming/adding/sunsetting flags and adding more docs to help reduce the ambiguity

- Marking `merge-node-identities` and `merge-service-identities` for deprecation and encouraging the use of their replacement `append-node-policy` and `append-service-identity`
- Updating docs to clarify that using `-rules` overwrites existing rules 
- Adding docs for the new flags: `append-node-policy` and `append-service-identity`

### PR Checklist

* [x] updated test coverage
* [x] external facing docs updated
* [x] not a security concern

### CODE REVIEW

File: command/acl/token/update/token_update.go [hunk 1]
```diff
@@ -79,9 +77,15 @@ func (c *cmd) init() {
 	c.flags.Var((*flags.AppendSliceValue)(&c.serviceIdents), "service-identity", "Name of a "+
 		"service identity to use for this token. May be specified multiple times. Format is "+
 		"the SERVICENAME or SERVICENAME:DATACENTER1,DATACENTER2,...")
+	c.flags.Var((*flags.AppendSliceValue)(&c.appendServiceIdents), "append-service-identity", "Name of a "+
```
Comment (kisunji) [inline command/acl/token/update/token_update.go:80]: can we add the "Overwrites existing X" comment on the non-append flags?

File: command/acl/token/update/token_update.go [hunk 2]
```diff
@@ -147,13 +155,39 @@ func (c *cmd) Run(args []string) int {
...
+	if hasAppendServiceFields {
+		parsedServiceIdents, err = acl.ExtractServiceIdentities(c.appendServiceIdents)
+	}
+
+	if hasAppendServiceFields && hasServiceFields {
+		c.UI.Error("Cannot combine the use of service-identity flag with append-service-identity. " +
+			"To set or overwrite existing service identities, use -service-identity. " +
+			"To append to existing service identities, use -append-service-identity.")
+		return 1
+	}
```
Comment (kisunji) [inline command/acl/token/update/token_update.go:170]: Let's move this up before we call `acl.ExtractServiceIdentities(c.serviceIdents)` to keep the validation ordering consistent with the NodeIdents below

File: command/acl/token/update/token_update_test.go [hunk 3]
```diff
@@ -109,6 +109,22 @@ func TestTokenUpdateCommand(t *testing.T) {
...
+			"-accessor-id=" + token.AccessorID,
+			"-token=root",
+			"-append-node-identity=third:node",
+			"-description=test token",
+		})
+
+		require.Len(t, token.NodeIdentities, 3)
+		require.Equal(t, "third", token.NodeIdentities[2].NodeName)
+		require.Equal(t, "node", token.NodeIdentities[2].Datacenter)
+	})
```
Comment (kisunji) [inline command/acl/token/update/token_update_test.go:126]: FYI subtests using `t.Run` should be independently runnable and pass. These append tests will not pass unless the previous subtests pass (but the previous test is also dependent on the previous-previous one).

As a team convention we tend to use `testutil.RunStep` to indicate a sequence of tests that must be run in a specific order.

Not a blocker, but just something to keep in mind as you write tests.

**Review by kisunji — APPROVED** [review]
Minor feedback but LGTM

**Review by cthain — APPROVED** [review]
No additional comments from me :+1:
